scipy/sparse/linalg/_matfuncs.py

```diff
@@ -861,3 +862,47 @@ def _ell(A, m):
     log2_alpha_div_u = np.log2(alpha/u)
     value = int(np.ceil(log2_alpha_div_u / (2 * m)))
     return max(value, 0)
+
+def matrix_power(A, power, structure=None):
+    """
+    Raise a square matrix to the integer power, `power`.
+
+    For non-negative integers, ``A**power`` is computed using repeated
+    matrix multiplications. Negative integers are not supported. 
+
+    Parameters
+    ----------
+    A : (M, M) square sparse array or matrix
+        sparse array that will be raised to power `power`
+    power : int
+        Exponent used to raise sparse array `A`
+
+    Returns
+    -------
+    A**power : (M, M) sparse array or matrix
+        The output matrix will be the same shape as A, and will preserve
+        the class of A, but the format of the output may be changed.
```

Comment:
Is it possible to know the output format in advance?

Reply:
Not quite. For example, if `A` is `bsr` or `csc`, then the power will be a `bsr` or `csc`(because `@` is stable in these formats), but if `A` is `coo` or `lil`, then the power will be `csr` (because `A@A` reformats `A` to `csr`). 

We could force format stability by saving the input format and re-formatting the output according to that? I think that may degrade performance unnecessarily though, and I don't think other areas make guarantees of format-stability?

Reply:
I think we better keep it as it is.